src/targets/gpu/hip.cpp

```diff
@@ -77,29 +77,70 @@ void* get_device_ptr(void* hptr)
     return result;
 }
 
-hip_ptr allocate_gpu(std::size_t sz, bool host = false)
+static auto& get_malloc_host_ptrs()
```

Comment:
I dont think we need two seperate maps here since we search both of them anyways.

Reply:
Removed.